src/csharp/Grpc.Core.Api/ChannelCredentialsConfiguratorBase.cs

```diff
@@ -0,0 +1,44 @@
+#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System.Collections.Generic;
+
+namespace Grpc.Core
+{
+    /// <summary>
+    /// Base class for objects that can consume configuration from <c>CallCredentials</c> objects.
+    /// Note: experimental API that can change or be removed without any prior notice.
+    /// </summary>
+    public abstract class ChannelCredentialsConfiguratorBase
```

Comment:
If SetSslCredentials has non-null values, e.g. not the default `new SslCredentials()`, then grpc-dotnet would raise a warning in its logs.

Reply:
Yes. 
grpc-dotnet would implement these: 
- the "default" SslCredentials() would be interpreted as standard https
- ChannelCredentials.Insecure would be interpreted as http
- composite channel credentials would be decomposed and checked that it's using the default SslCredentials and the call credentials will be used for all calls on a given channel.

All other configurations could just throw as "unsupported" (it needs to actually fail, not just log an error, otherwise this would be a security problem). In the future, we could also implement functionality that converts the SslCredentials arguments into X509 certificate objects and passes them to the HTTPClient.